### ISSUE TITLE
UTF-8 and Optionals

### ISSUE THREAD

**Issue by kirsle** [aichaos/rivescript-python#37]
This might be related to [rivescript-js#147](https://github.com/aichaos/rivescript-js/issues/147). The following test case produces the problem:

```
+ *
- Häää?

+ [*] (überrasch mich|empfiehl mir was|empfehl mir was|was gibts neues für mich|empfehlung) [*]
- recommendation
```

The words with umlauts in them make the second trigger unmatchable.

In rivescript-js#147 I found Python wasn't affected by the Chinese symbols but may be having problems with these umlauts for some reason.

**Comment by Xiangshen-Meng** [comment]
Maybe the Chinese symbols problem is caused by space.
https://github.com/aichaos/rivescript-python/blob/master/rivescript/brain.py#L449
For language doesn't separate by space(Like Chinese), this line may cause problem.
``` python
p = r'(?:\\s|\\b)+{}(?:\\s|\\b)+'.format(p)
```
How about change it to this?

``` python
p = r'(?:\\s|\\b)?{}(?:\\s|\\b)?'.format(p)
```

But this may cause the English in word options.

Sorry I am not familiar with RiveScript and this is only my thoughts.

**Comment by kirsle** [comment]
Fixed by PR #108
